Remove UnaryServerInterceptor (#7115)
open-telemetry/opentelemetry-go-contrib: CHANGELOG.md

- The deprecated `SemVersion` function is removed in `go.opentelemetry.io/contrib/instrumentation/github.com/gin-gonic/gin/otelgin`, use `Version` function instead. (#7085)
- The deprecated `SemVersion` function is removed in `go.opentelemetry.io/contrib/instrumentation/github.com/gin-gonic/gin/otelgin/test`, use `Version` function instead. (#7087)
- The deprecated `SemVersion` function is removed in `go.opentelemetry.io/contrib/instrumentation/github.com/labstack/echo/otelecho`, use `Version` function instead. (#7089)

> - The deprecated `UnaryServerInterceptor` from `go.opentelemetry.io/contrib/instrumentation/google.golang.org/grpc/otelgrpc` module is removed. (#7115)

Reviewer: ```suggestion
- The deprecated `UnaryServerInterceptor` in `go.opentelemetry.io/contrib/instrumentation/google.golang.org/grpc/otelgrpc` is removed. (#7115)
```
